service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java

```diff
@@ -0,0 +1,247 @@
+package com.codeforcommunity.processor;
+
+import static org.jooq.generated.Tables.ADOPTED_SITES;
+import static org.jooq.generated.Tables.SITES;
+import static org.jooq.generated.Tables.SITE_ENTRIES;
+import static org.jooq.generated.Tables.STEWARDSHIP;
+
+import com.codeforcommunity.api.IProtectedSiteProcessor;
+import com.codeforcommunity.auth.JWTData;
+import com.codeforcommunity.dto.site.AddSiteRequest;
+import com.codeforcommunity.dto.site.AdoptedSitesResponse;
+import com.codeforcommunity.dto.site.RecordStewardshipRequest;
+import com.codeforcommunity.dto.site.UpdateSiteRequest;
+import com.codeforcommunity.enums.PrivilegeLevel;
+import com.codeforcommunity.exceptions.AuthException;
+import com.codeforcommunity.exceptions.ResourceDoesNotExistException;
+import com.codeforcommunity.exceptions.WrongAdoptionStatusException;
+import java.sql.Timestamp;
+import java.util.List;
+import org.jooq.DSLContext;
+import org.jooq.generated.tables.records.AdoptedSitesRecord;
+import org.jooq.generated.tables.records.SiteEntriesRecord;
+import org.jooq.generated.tables.records.SitesRecord;
+import org.jooq.generated.tables.records.StewardshipRecord;
+
+public class ProtectedSiteProcessorImpl implements IProtectedSiteProcessor {
+
+  private final DSLContext db;
+
+  public ProtectedSiteProcessorImpl(DSLContext db) {
+    this.db = db;
+  }
+
+  private void checkSiteExists(int siteId) {
+    if (!db.fetchExists(db.selectFrom(SITES).where(SITES.ID.eq(siteId)))) {
+      throw new ResourceDoesNotExistException(siteId, "Site");
+    }
+  }
+
+  private Boolean isAlreadyAdopted(int userId, int siteId) {
```

Comment:
Just for clarification, a site can be adopted by multiple users? If that's true, then maybe change the method name here to `isAlreadyAdoptedByUser()` to make it more clear.

Reply:
No, a site can only be adopted by 1 person. Added the isAlreadyAdoptedByUser method for unadopting.